packages/api/lambdas/replay-archived-s3-messages.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const pMap = require('p-map');
+
+const { envUtils } = require('@cumulus/common');
+const { getJsonS3Object } = require('@cumulus/aws-client/S3');
+const { s3 } = require('@cumulus/aws-client/services');
+const { getQueueUrlByName, sendSQSMessage } = require('@cumulus/aws-client/SQS');
+const { getS3PrefixForArchivedMessage } = require('@cumulus/ingest/sqs');
+const Logger = require('@cumulus/logger');
+
+const logger = new Logger({ sender: '@cumulus/replay-archived-messages' });
+
+// Get messages from S3 using queueName
+const getArchivedMessagesFromQueue = async (queueName) => {
+  let listObjectsResponse;
+  let continuationToken;
+  const bucket = envUtils.getRequiredEnvVar('system_bucket', process.env);
+  const stackName = envUtils.getRequiredEnvVar('stackName', process.env);
+  const prefix = getS3PrefixForArchivedMessage(stackName, queueName);
+  const params = {
+    Bucket: bucket,
+    Prefix: prefix,
+    ContinuationToken: continuationToken,
+    Delimiter: '/',
+  };
+  logger.debug(`Params for listS3Keys bucket: ${bucket}, prefix: ${prefix}`);
+  const archivedMessages = [];
+
+  /* eslint-disable no-await-in-loop */
+  do {
```

Comment:
We should also add a unit test for this function to prove that even if one message errors (because it's not a JSON file), the other messages are returned

Reply:
[`1cd1adc` (#2373)](https://github.com/nasa/cumulus/pull/2373/commits/1cd1adc0f360abf13607d262680fdb73d1a0d6a1)